### PR TITLE
Initial check-in for Getting Started guide for 0.2

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

A set of documentation updates to refresh information for the 0.2 release

## Changes

- Fixes: #670
- [X] Getting started
- [X] Installation
- [X] Configuration Guide
- [X] FAQ for common runtime queries
- [X] Examples documentation
- [X] Docs per platform (contained in \Documentation~ folder)

### CODE REVIEW

File: articles/systems/01-camera-system.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ For the most part, you should not need to alter the camera system configuration
 
 ### Related Articles
 
-* [XRTK Platform System](08-platform-system.md)
+* [XRTK Platform System](08-platform-framework.md)
```
Comment (StephenHodgson) [inline articles/systems/01-camera-system.md:41]: Still not sure about the name of this article as there isn't a platform _system_
```suggestion
* [XRTK Platform framework](08-platform-framework.md)
```

File: articles/systems/08-platform-framework.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Mixed Reality Toolkit Platform Framework
+
+A major component of the Mixed Reality Toolkit is the Platform Framework that was employed to both accurately determine what platforms have been added to the scope of the project and also to determine when a platform is active/running.
```
Comment (StephenHodgson) [inline articles/systems/08-platform-framework.md:3]:
```suggestion
A major component of the Mixed Reality Toolkit is the Platform definitions that was employed to both accurately determine what platforms have been added to the scope of the project and also to determine when a platform is active/running.
```

File: articles/systems/08-platform-framework.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Mixed Reality Toolkit Platform Framework
```
Comment (StephenHodgson) [inline articles/systems/08-platform-framework.md:1]:
```suggestion
# Mixed Reality Toolkit Platforms
```

File: articles/systems/08-platform-framework.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+The list support multiple selections (for cases where a system or profile) can operate for multiple platforms.
+
+The list is dynamic and will update based on which XRTK platforms have been installed and what platforms are supported by the developers Unity installation.
+
+## Platform Identification
+
+To explain how this system basically operates, there is a `Platform` script in the UPM package runtime folder which implements the required interfaces for the platform system, including:
+
+```csharp
+    public override bool IsAvailable;
```
Comment (StephenHodgson) [inline articles/systems/08-platform-framework.md:20]: why duplicate what's already written in code?

if this was to change in the future it'd be high upkeep

File: articles/systems/08-platform-framework.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+`IsAvailable` indicates whether the platform is available at runtime, usually checking for a vendor SDK or other API call which signifies the platform is available.
```
Comment (StephenHodgson) [inline articles/systems/08-platform-framework.md:23]: I'd rather see this documented in the `BasePlatform` or `IMixedRealityPlatform` documentation in the class itself

File: articles/systems/08-platform-framework.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+It also includes a set of `PlatformOverrides` which help to specify which targets a platform should be available for by default.
+
+## Platform Default Configuration
+
+Included with each platform, a default platform configuration asset is provided to help automatically deploy the necessary configuration to operate a platform, usually including the Camera System and COntroller configuration, but can include others as required.  This is located within the `Profiles` folder of the `XRTK.Generated` folder for the specific platform, for example:
+
+![](../images/Configuation/Platforms/PlatformConfigurationDefaults.png)
+
+Once selected, the default configuration will appear in the inspector and a `` button is available to install the configuration in to your current scene.
```
Comment (StephenHodgson) [inline articles/systems/08-platform-framework.md:39]: Missing something here?

File: articles/00-GettingStarted.md [hunk 7]
```diff
@@ -2,48 +2,66 @@
...
 ---
 
-### Adding the Mixed Reality Toolkit to your project
+## Adding the Mixed Reality Toolkit to your project
 
-The Mixed Reality Toolkit is available via [multiple delivery mechanisms](01-DownloadingTheXRTK.md), primarily via the Unity Package Management system.
+First download the Mixed Reality Toolkit via one of the [delivery mechanisms](01-DownloadingTheXRTK.md) into your Unity project.
 
-Our preferred deployment approach is to fully utilize [Unity's native package manager](https://docs.unity3d.com/Manual/upm-ui.html) to incorporate the Mixed Reality Toolkit in your solution, akin to the other modules Unity provides out of the box.  This is the quickest and safest way to get Mixed Reality Toolkit in your solution and is automatically updated as new releases are published.
+Our preferred deployment approach is to use the [Unity Package Manager system](01-DownloadingTheXRTK.md#register-upm-server-in-project-settings).  This is the quickest and safest way to get Mixed Reality Toolkit in your solution and provides new releases as soon as they are published. Using UPM, there are also no asset conflicts in your Unity project.
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:52]:
```suggestion
Our preferred deployment approach is to use the [Unity Package Manager](01-DownloadingTheXRTK.md#register-upm-server-in-project-settings).  This is the quickest and safest way to get Mixed Reality Toolkit in your solution and provides new releases as soon as they are published. Using UPM, there are also no asset conflicts in your Unity project.
```

File: articles/00-GettingStarted.md [hunk 8]
```diff
@@ -2,48 +2,66 @@
...
 
 > **Note:** Some prefabs and assets require TextMesh Pro, meaning you have to have the TextMesh Pro package installed and the assets in your project `Window -> TextMeshPro -> Import TMP Essential Resources`.
 
 Simply follow along with the UPM Installation instructions detailed in the [Downloading the XRTK](01-DownloadingTheXRTK.md#register_upm_server_in_project_settings) guide to add the XRTK scoped registry and install the relevant packages.
 
-> We recommend installing the **SDK** first to ensure you have the smoothest setup with the XRTK and give you access to the Quickstart configurations.
+### It all starts with the SDK
+
+We recommend installing the **XRTK.SDK** first to ensure you have the smoothest setup with the XRTK, which also gives you access to the Quickstart configurations.  **DO NOT install any additional platforms until the SDK is installed and your starting scene is configured**
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:60]: Seems complicated to get started

File: articles/00-GettingStarted.md [hunk 9]
```diff
@@ -53,11 +71,11 @@ Configuring your scene is extremely simple by simply selecting the following fro
 
 > `Mixed Reality Toolkit -> Configure...`
 
-![Configure](../images/GettingStarted/MixedRealityConfigure.png)
+![Configure](/images/GettingStarted/MixedRealityConfigure.png)
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:74]: What's the difference here? The links already work and I'm afraid the actual html docs build will break without this prefix

File: articles/00-GettingStarted.md [hunk 10]
```diff
@@ -66,32 +84,94 @@ Which contains the following:
...
+
+When building for Android, there is an additional requirement to install an `AndroidManaifest` for the quest to operate correctly, the XRTK provides a fast an easy way to do this by simply clicking `Mixed Reality Toolkit -> Tools -> Oculus -> Create Oculus Quest Compatible AndroidManaifest.xml` in the Unity editor menu. (**without this, your project may not start or appear in 2D**)
+
+Please ensure the `Oculus` XRTK package is installed and that the Oculus Camera and controller providers are configured.
+
+### Building for Magic Leap
+
+Magic Leap is the one exception that does require the new Unity XR Management system and will require a separate Unity project at this time, primarily due the way the Magic leap platform is evolving.  We recommend building your project for one of the other platforms and then simply copying the asset folder to a new Unity project for Magic Leap to build and run.
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:159]: >  will require a separate Unity project at this time

What does this even mean? I'm not sure where this information came from.

>  We recommend building your project for one of the other platforms and then simply copying the asset folder to a new Unity project for Magic Leap to build and run.

Why?

File: articles/00-GettingStarted.md [hunk 11]
```diff
@@ -66,32 +84,94 @@ Which contains the following:
...
+
+Please ensure the `Windows Mixed Reality` XRTK package is installed and that the Windows Mixed Reality Camera and controller providers are configured.
+
+### Building for Oculus Quest / Android
+
+Building for Oculus Quest is rather unique, if you have your Oculus Quest connected to your machine via a high speed USB3 cable and have the Oculus Client running, you can run Quest project in Windows Standalone mode directly on the Quest via Oculus Link, [please see Oculus Support for further details](https://support.oculus.com/444256562873335/).  This will enable both controller and hands support for quick development.
+
+When building for Android, there is an additional requirement to install an `AndroidManaifest` for the quest to operate correctly, the XRTK provides a fast an easy way to do this by simply clicking `Mixed Reality Toolkit -> Tools -> Oculus -> Create Oculus Quest Compatible AndroidManaifest.xml` in the Unity editor menu. (**without this, your project may not start or appear in 2D**)
+
+Please ensure the `Oculus` XRTK package is installed and that the Oculus Camera and controller providers are configured.
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:155]: A lot of this seems repetitive per platform, couldn't we word this in a more generic way?

File: articles/00-GettingStarted.md [hunk 12]
```diff
@@ -66,32 +84,94 @@ Which contains the following:
...
+* Unity Magic Leap XR package installed
+* Magic Leap Unity asset installed (from Magic Leap hub)
+* Magic Leap Developer Hub
+* Visual Studio 2019
 
-Once the build completes, you can install and run it on the device.
+Please ensure the `Lumin` XRTK package is installed and that the Lumin Camera and controller providers are configured.
 
-> **Note:** The build window automatically installs some platforms for you, but depending on the platform, you'll need to follow the platform specific installation steps to get it manually installed on your device.
+**A more detailed guide for building for Magic leap will follow soon**
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:174]: I think this article already gets too much into platform specific setup requirements. It might be better to break out each platform specific setup into their own articles and keep this one as generic as possible with links to platform specific articles as needed.

File: articles/01-DownloadingTheXRTK.md [hunk 13]
```diff
@@ -19,11 +19,11 @@ Unity has now enabled the ability to register scoped registries through the Unit
 
 To start, you will need to register the XRTK Scoped registry in the Unity Package Manager, which you will find in the Package Manager Window (Window -> Package Manager) Advanced dropdown, as shown below:
 
-![Unity Package Manager Advanced options](../images/DownloadingTheXRTK/UnityPackageManager-Advanced.jpg)
+![Unity Package Manager Advanced options](/images/DownloadingTheXRTK/UnityPackageManager-Advanced.jpg)
```
Comment (StephenHodgson) [inline articles/01-DownloadingTheXRTK.md:22]: Why this change?

File: articles/02-Configuration.md [hunk 14]
```diff
@@ -1,50 +1,48 @@
...
-![Clone Profile Button](../images/Configuation/MixedRealityProfileCloneButton.png)
+![Clone Profile Button](/images/Configuration/MixedRealityProfileCloneButton.png)
 
 ---
 
 ### Related Articles
 
 * [Getting Started](00-GettingStarted.md#getting-started-with-the-mixed-reality-toolkit)
-  * [Building and running your mixed reality application](00-GettingStarted.md#build-and-play)
+* [Building and running your mixed reality application](00-GettingStarted.md#build-and-play)
```
Comment (StephenHodgson) [inline articles/02-Configuration.md:45]: Let's revert this. It is indented bc it's a subheading in the main article

File: articles/05-FAQ.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
+# Mixed Reality Toolkit (XRTK) Frequently Asked Questions
+
+![The Mixed Reality Toolkit](/images/Branding/XRTK_Logo_1200x250.png)
+
+## 1. How do I create a new Mixed Reality Toolkit Project?
+
+We recommend walking through the [Getting Started guide](00-GettingStarted.md) for a detailed understanding to get up and running.
```
Comment (StephenHodgson) [inline articles/05-FAQ.md:15]: Why duplicate text here instead of just providing links to the specific articles?

File: articles/platforms/webxr.md [hunk 16]
```diff
@@ -0,0 +1,26 @@
+# Headline
+
+A brief overview of the tutorial.
```
Comment (StephenHodgson) [inline articles/platforms/webxr.md:3]: not complete?

File: articles/systems/02-input-system.md [hunk 17]
```diff
@@ -1,23 +1,125 @@
...
+* Removing controllers that are not needed in [Controller Definitions](#controller-definitions)
 
 ---
 
 ### Related Articles
 
-* [Article 1]()
-* [Article 2]()
-* [Article 3]()
+* [XRTK Platform System](08-platform-framework.md)
```
Comment (StephenHodgson) [inline articles/systems/02-input-system.md:122]: I'd like to see links back to articles that reference this one (like getting started?)

File: articles/systems/03-boundary-system.md [hunk 18]
```diff
@@ -1,23 +1,30 @@
...
+
+* Display settings - Options to show the boundary (or not), as well as its height, physics layer and default material.
+* Floor settings - Options to show the floor (or not), as well as the material to be used for the floor.
+* Wall settings - Options to show the boundary walls (or not), as well as the material to be used for the walls.
+* Ceiling settings - Options to show the ceiling (or not), as well as the material to be used for the ceiling.
 
-### Related Articles
+## Further notes
 
-* [Article 1]()
```
Comment (StephenHodgson) [inline articles/systems/03-boundary-system.md:18]: I'd like to see links back to articles that reference this one (like getting started?)

File: articles/systems/04-teleporting-system.md [hunk 19]
```diff
@@ -1,23 +1,53 @@
...
+For the most part, you should not need to alter the Teleport system configuration unless required.
 
 ---
 
 ### Related Articles
 
-* [Article 1]()
-* [Article 2]()
-* [Article 3]()
+* [XRTK Platform System](08-platform-framework.md)
```
Comment (StephenHodgson) [inline articles/systems/04-teleporting-system.md:50]: I'm not sure platform stuff needs to be linked at the bottom of all these systems. It's not really related

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
Tried building the docs locally but the links to images are now broken

File: articles/systems/01-camera-system.md (hunk 1 above)
Comment (SimonDarksideJ) [reply articles/systems/01-camera-system.md]: All references to Platform System updated to Platform Framework

File: articles/systems/08-platform-framework.md (hunk 4 above)
Comment (SimonDarksideJ) [reply articles/systems/08-platform-framework.md:20]: As it's new and we are introducing it, it is worthwhile to highlight it. We can always update or remove this section later as we evolve.

File: articles/systems/08-platform-framework.md (hunk 5 above)
Comment (SimonDarksideJ) [reply articles/systems/08-platform-framework.md:23]: This section is purely a highlight of that area and the advances we have made.  That documentation should already be in the API auto-generated documentation.  We can evolve that further in 0.3

File: articles/00-GettingStarted.md (hunk 10 above)
Comment (SimonDarksideJ) [reply articles/00-GettingStarted.md:159]: You cannot have the legacy XR system AND the new XR Plugin Management system enabled at the same time. 
Unless you are saying that Magic leap still works with the legacy XR system only?

File: articles/00-GettingStarted.md (hunk 11 above)
Comment (SimonDarksideJ) [reply articles/00-GettingStarted.md:155]: Some of it is, and we can evolve it later.  However, each doc is unique in itself and we cannot guarantee readers will be willing to hope around.  GettingStarted is meant to be standalone in effect.

Experience has taught me that most readers will only read the section they are interested in and not the whole doc

File: articles/00-GettingStarted.md (hunk 12 above)
Comment (SimonDarksideJ) [reply articles/00-GettingStarted.md:174]: I have created individual Platform documents with specifics and a lot more highlights.  These are just the short versions for the getting started guide.

But as I stated, I needed assistance to update the Magic Leap version

File: articles/02-Configuration.md (hunk 14 above)
Comment (SimonDarksideJ) [reply articles/02-Configuration.md:45]: I'd disagree here, just because it's indented in the TOC doesn't mean it needs indenting here.

Everywhere else the "Related Links" section is just a list of links, regardless of how they are laid out in the TOC

File: articles/05-FAQ.md (hunk 15 above)
Comment (SimonDarksideJ) [reply articles/05-FAQ.md:15]: As the FAQ explains, it's a short short version for those who don't want to follow the full guide.  One of the feedbacks from the recent "complexities" with the old project was it was too hard to find stuff and took to much reading to get started.

This isn't a replacement, just a quick "how to"

File: articles/00-GettingStarted.md (hunk 9 above)
Comment (SimonDarksideJ) [reply articles/00-GettingStarted.md]: Image linkds fixed

File: articles/01-DownloadingTheXRTK.md (hunk 13 above)
Comment (SimonDarksideJ) [reply articles/01-DownloadingTheXRTK.md]: Links fixed

File: articles/platforms/webxr.md (hunk 16 above)
Comment (SimonDarksideJ) [reply articles/platforms/webxr.md]: Updated

File: articles/systems/02-input-system.md (hunk 17 above)
Comment (SimonDarksideJ) [reply articles/systems/02-input-system.md]: Added

File: articles/00-GettingStarted.md (hunk 10 above)
Comment (StephenHodgson) [reply articles/00-GettingStarted.md:163]: Depends on which XR Plugin version you install

File: articles/02-Configuration.md (hunk 14 above)
Comment (StephenHodgson) [reply articles/02-Configuration.md:45]: I just want it to follow the article body. 

* An h1 heading
  * An h2 heading
    * An h3 heading

File: articles/00-GettingStarted.md [hunk 20]
```diff
@@ -2,48 +2,66 @@
...
 
 > **Note:** Some prefabs and assets require TextMesh Pro, meaning you have to have the TextMesh Pro package installed and the assets in your project `Window -> TextMeshPro -> Import TMP Essential Resources`.
 
 Simply follow along with the UPM Installation instructions detailed in the [Downloading the XRTK](01-DownloadingTheXRTK.md#register_upm_server_in_project_settings) guide to add the XRTK scoped registry and install the relevant packages.
 
-> We recommend installing the **SDK** first to ensure you have the smoothest setup with the XRTK and give you access to the Quickstart configurations.
+### It all starts with the SDK
+
+We recommend installing the **XRTK.SDK** first to ensure you have the smoothest setup with the XRTK, which also gives you access to the Quickstart configurations.  **DO NOT install any additional platforms until the SDK is installed and your starting scene is configured**
```
Comment (StephenHodgson) [inline articles/00-GettingStarted.md:60]:
```suggestion
We recommend installing the **XRTK.SDK** first to ensure you have the smoothest setup with the XRTK, which also gives you access to the Quickstart configurations.  **DO NOT install any additional platforms until the SDK is installed**
```

File: articles/00-GettingStarted.md (hunk 11 above)
Comment (StephenHodgson) [reply articles/00-GettingStarted.md:157]: Fine but I'm personally unlikely to keep the docs up to date. I just wanna minimize the overhead of this task.

File: articles/toc.yml [hunk 21]
```diff
@@ -23,4 +46,6 @@
     - name: Coding Guidelines
       href: appendices/A01-CodingGuidelines.md
     - name: XRTK vs MRTK
-      href: appendices/A02-ChoosingXRTKvsMRTK.md
\ No newline at end of file
+      href: appendices/A02-ChoosingXRTKvsMRTK.md
+    - name: XRTK Public Roadmap
+      href: appendices/A03-Roadmap.md
```
Comment (StephenHodgson) [inline articles/toc.yml:51]: Should we also add known issues to the appendix?

File: articles/systems/01-camera-system.md [hunk 22]
```diff
@@ -1,23 +1,44 @@
...
+The Camera system for the XRTK allows for different implementations to be utilized for different runtime or build platforms. Each platform provided by the XRTK includes a default camera system configuration for that platform which can be customized if you wish:
+
+![](../../images/Configuration/CameraSystem/CameraSystemSettings.png)
+
+The configuration holds:
+
+* The default configuration - a Camera profile to be used where none is specified, a fallback default
+* Additional Camera Profile variants, which specifies the Name, Type, Camera Profile and platforms this variant is meant to run on.
+
+> For more detail on the [Platforms](08-platform-framework.md) that are available to configure, see the [XRTK Platform Framework](08-platform-framework.md) documentation.
```
Comment (StephenHodgson) [inline articles/systems/01-camera-system.md:24]: Why is `08-platform-framework.md` linked twice?

File: articles/00-GettingStarted.md (hunk 11 above)
Comment (SimonDarksideJ) [reply articles/00-GettingStarted.md:157]: That's where I come in 😂

File: articles/systems/01-camera-system.md (hunk 22 above)
Comment (SimonDarksideJ) [reply articles/systems/01-camera-system.md:24]: More the merrier 😉

**Review by StephenHodgson — APPROVED** [review]
